apps/system/js/bluetooth_v2.js

```diff
@@ -0,0 +1,430 @@
+    this._adapter = this._bluetooth.defaultAdapter;
+    if (this._adapter) {
+      // dispatch event when default adapter is available
+      this._dispatchAdapterState(true);
+
+      // dispatch default bluetooth enable state
+      this._isEnabled = (this._adapter.state === 'enabled');
+      this._dispatchEnableState();
+    } else {
+      // dispatch event when default adapter is not available
+      this._dispatchAdapterState(false);
+    }
+
+    // hold the instance so we can remove it when needed
+    this._btDefaultAdapterChangeHandler =
+      this._btManagerHandler.bind(this, null);
+    this._bluetooth.addEventListener('attributechanged',
+      this._btDefaultAdapterChangeHandler);
+
+    // decouple enable/disable bluetooth function from other system part
+    window.addEventListener('request-enable-bluetooth',
+      this._requestEnableHandler.bind(this));
+    window.addEventListener('request-disable-bluetooth',
+      this._requestDisableHandler.bind(this));
+
+    /*
+     * In file transfering case:
+     * since System Message can't be listened in two js files within a app,
+     * so we listen here but dispatch events to bluetooth_transfer.js
+     * while getting bluetooth file transfer start/complete system messages
+     */
+    navigator.mozSetMessageHandler('bluetooth-opp-transfer-start',
+      (transferInfo) => {
+        this._setProfileConnected(this.Profiles.OPP, true);
+        window.dispatchEvent(new CustomEvent('bluetooth-opp-transfer-start',
+          {transferInfo: transferInfo}));
+      }
+    );
+
+    navigator.mozSetMessageHandler('bluetooth-opp-transfer-complete',
+      (transferInfo) => {
+        this._setProfileConnected(this.Profiles.OPP, false);
+        window.dispatchEvent(new CustomEvent('bluetooth-opp-transfer-complete',
+          {transferInfo: transferInfo}));
+      }
+    );
+
+    Service.registerState('isEnabled', this);
+    // LazyLoader.load(['js/bluetooth_icon.js',
+    //                  'js/bluetooth_transfer_icon.js',
+    //                  'js/bluetooth_headphone_icon.js'], function() {
+    //   this.icon = new BluetoothIcon(this);
+    //   this.icon.start();
+    //   this.transferIcon = new BluetoothTransferIcon(this);
+    //   this.transferIcon.start();
+    //   this.headphoneIcon = new BluetoothHeadphoneIcon(this);
+    //   this.headphoneIcon.start();
+    // }.bind(this)).catch(function(err) {
+    //   console.error(err);
+    // });
+  },
+
+  /**
+   * Once adapter is available, listen to bluetooth state and
+   * the connect stat of supported profiles.
+   *
+   * we only support two use cases for bluetooth connection:
+   *   1. connecting with a headset
+   *   2. transfering a file to/from another device
+   * So we need to listen to corresponding events to know we are (aren't)
+   * connected, then summarize to an event and dispatch to StatusBar
+   *
+   * @private
+   */
+  _registerListeners: function bt__registerListeners() {
```

Comment:
Again. Naming. Please give some readable name to be relative with connection profiles change. It's hard to understand `_registerListeners` in `start` function since there are many events listened there.